micropython/ssa/core.py

```
@@ -157,25 +156,57 @@ def __connect(self, last_will: str | None = None, _with_registration: bool = Fal
+            f.write(binary.decode("utf-8"))
 
         print("[INFO] Firmware write complete. Restarting device.")
-        machine.soft_reset()
+        from machine import soft_reset
+        soft_reset()
 
-    def __handle_config_change(self, config: str):
+    def __handle_user_config(self, config: str):
         raise Exception(f"[TODO] ssa.__handle_config_change not implemented")
 
+    def register_handler(self, task: function):
```

Ruff (0.8.2)

Undefined name `function`

(F821)